src/utils/viewer/viewer.cpp

```diff
@@ -737,62 +669,53 @@ void Viewer::mouseMoveEvent(MouseMoveEvent& event) {
 // NOTE: Mouse + shift is to select object on the screen!!
 void Viewer::keyPressEvent(KeyEvent& event) {
   const auto key = event.key();
-  bool agentMoved = false;
   switch (key) {
     case KeyEvent::Key::Esc:
       std::exit(0);
       break;
     case KeyEvent::Key::Left:
-      controls_(*agentBodyNode_, "turnLeft", lookSensitivity);
-      agentMoved = true;
+      simulator_->getAgent(defaultAgentId_)->act("turnLeft");
```

Comment:
Sorry, my bad. In case you did not see my previous comments, post it again:
You can consider caching the defaultAgent_ instead of the defaultAgentId_. because you need to call it every time below.